packages/components/file-button/src/file-button.tsx

```diff
@@ -54,102 +60,96 @@ export type FileButtonProps = Omit<ButtonProps, "onChange" | "children"> &
-        >
-          {children}
-        </Component>
-      ) as ReactNode
-    }
-
-    assignRef(resetRef, onReset)
-
-    return (
-      <>
-        <ui.input
-          ref={mergeRefs(inputRef, ref)}
-          type="file"
-          aria-hidden
-          tabIndex={-1}
-          id={id}
-          name={name}
-          form={form}
-          accept={accept}
-          multiple={multiple}
-          style={{
-            border: "0px",
-            clip: "rect(0px, 0px, 0px, 0px)",
-            height: "1px",
-            width: "1px",
-            margin: "-1px",
-            padding: "0px",
-            overflow: "hidden",
-            whiteSpace: "nowrap",
-            position: "absolute",
-          }}
-          onChange={onChange}
-          aria-readonly={ariaReadonly}
-          {...pickObject(rest, formControlProperties)}
-        />
-
-        {isFunction(children)
-          ? children({
-              onClick,
-              disabled,
-              readOnly,
-              required,
-              isDisabled: disabled,
-              isReadOnly: readOnly,
-              isRequired: required,
-              isInvalid,
-            })
-          : children}
-      </>
-    )
-  },
-)
+export const FileButton = forwardRef<FileButtonProps, "input">((props, ref) => {
+  const {
+    id,
+    name,
+    accept,
+    multiple,
+    form,
+    "aria-readonly": ariaReadonly,
+    onClick: onClickProp,
+    onChange: onChangeProp,
+    ...rest
+  } = useFormControlProps(props)
+
+  const [styles, mergedProps] = useMultiComponentStyle("FileButton", {
+    ...props,
+  })
+
+  let { className, resetRef, as: As, children } = omitThemeProps(mergedProps)
+
+  const { disabled, readOnly, required, "aria-invalid": isInvalid } = rest
+
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const onClick = useCallback(() => {
+    if (disabled || readOnly) return
+
+    inputRef.current?.click()
+  }, [disabled, readOnly])
+
+  const onChange = useCallback(
+    (ev: ChangeEvent<HTMLInputElement>) => {
+      const files = !isNull(ev.currentTarget.files)
+        ? Array.from(ev.currentTarget.files)
+        : undefined
+
+      onChangeProp?.(files)
+    },
+    [onChangeProp],
+  )
+
+  const onReset = useCallback(() => {
+    if (inputRef.current) inputRef.current.value = ""
+  }, [])
+
+  if (!isFunction(children)) {
+    const Component = As || Button
+
+    children = (
+      <Component
+        className={cx("ui-file-button", className)}
+        {...rest}
+        onClick={handlerAll(onClickProp, onClick)}
+      >
+        {children}
+      </Component>
+    ) as ReactNode
+  }
+
+  assignRef(resetRef, onReset)
+
+  return (
+    <>
+      <ui.input
+        ref={mergeRefs(inputRef, ref)}
+        type="file"
+        aria-hidden
+        tabIndex={-1}
+        id={id}
+        name={name}
+        form={form}
+        accept={accept}
+        multiple={multiple}
+        style={styles}
```

Comment:
This should not be themed.
There is no need to tamper with this style in the theme, as it is intended to be invisible.

Reply:
Do I need to just delete these code? I deleted them.